Sources/Commands/Describe.swift

```diff
@@ -10,85 +10,354 @@
-        ])
+        init(from dependency: PackageDependencyDescription) {
+            self.name = dependency.explicitName
+            self.url = dependency.url
+            self.requirement = dependency.requirement
+        }
     }
-}
 
-extension Target: JSONSerializable {
-
-    func describe(on stream: OutputByteStream, indent: Int = 0) {
-        stream <<< Format.asRepeating(string: " ", count: indent)
-            <<< "Name: " <<< name <<< "\n"
-        stream <<< Format.asRepeating(string: " ", count: indent)
-            <<< "C99name: " <<< c99name <<< "\n"
-        stream <<< Format.asRepeating(string: " ", count: indent)
-            <<< "Type: " <<< type.rawValue <<< "\n"
-        stream <<< Format.asRepeating(string: " ", count: indent)
-            <<< "Module type: " <<< String(describing: Swift.type(of: self)) <<< "\n"
-        stream <<< Format.asRepeating(string: " ", count: indent)
-            <<< "Path: " <<< sources.root.pathString <<< "\n"
-        stream <<< Format.asRepeating(string: " ", count: indent)
-            <<< "Sources: " <<< sources.relativePaths.map({ $0.pathString }).joined(separator: ", ") <<< "\n"
+    /// Represents a product for the sole purpose of generating a description.
+    struct DescribedProduct: Encodable {
+        let name: String
+        let type: ProductType
+        let targets: [String]
+
+        init(from product: Product, in package: Package) {
+            self.name = product.name
+            self.type = product.type
+            self.targets = product.targets.map { $0.name }
+        }
     }
 
-    public func toJSON() -> JSON {
-        return .init([
-            "name": name,
-            "c99name": c99name,
-            "type": type,
-            "module_type": String(describing: Swift.type(of: self)),
-            "path": sources.root,
-            "sources": sources
-        ])
+    /// Represents a target for the sole purpose of generating a description.
+    struct DescribedTarget: Encodable {
+        let name: String
+        let type: String
+        let c99name: String?
+        let moduleType: String?
+        let path: String
+        let sources: [String]
+        let resources: [PackageModel.Resource]?
+        
+        init(from target: Target, in package: Package) {
+            self.name = target.name
+            self.type = target.type.rawValue
+            self.c99name = target.c99name
+            self.moduleType = String(describing: Swift.type(of: target))
+            self.path = target.sources.root.relative(to: package.path).pathString
+            self.sources = target.sources.relativePaths.map{ $0.pathString }
+            self.resources = target.resources.isEmpty ? nil : target.resources
+        }
     }
 }
 
-extension Sources: JSONSerializable {
-    public func toJSON() -> JSON {
-        return .array(relativePaths.map({ .string($0.pathString) }))
+
+public struct PlainTextEncoder {
```

Comment:
This is pretty cool. Do you think it would make sense to move this to TSCBasic?

Reply:
I considered that, but I think we'd have to weigh the trade-off of making it generic enough vs the value.  I think it should be kept private for now since there are some things that are fairly specific to this format, but it definitely seems like the kind of thing that could be generalized with more work.

Reply:
cc @neonichu, this may be useful in the collections CLI work

Reply:
True enough.  I'd suggest doing the work to extract it and generalize it in a follow-on PR.

Reply:
I agree with your above statement that we should not eagerly-generalize and move things to TSC before they are mature. In this case, lets check if this is useful to @neonichu and if so we can put it in Basics and let it bake there for a while

Reply:
Yah, this looks useful. I'll try it out and move it to `Basics` as part of CLI PR if it works out.

Reply:
Great, thanks.  There is still some unfinished business about controlling the output formatting, but we can collaborate on adding ways to control that.